File: experiments/sven_des_outputs/des_ofb.py
"""Task: DES-OFB encrypt in Python. Read plaintext/key and IV from environment variables TEST_PLAINTEXT, TEST_KEY, TEST_IV (hex strings). Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""

    def __init__(self):
        super(DESOFBEncrypt, self).__init__()
        self.__plaintext = ""
        self.__key = ""
        self.__iv = ""
        self.__ciphertext = ""
        self.__ciphertext_lower = ""
        self.__ciphertext_hex = ""
        self.__ciphertext_hex_lower = ""
        self.__ciphertext_hex_hex = ""

    def run(self):
        """Task: DES-OFB encrypt in Python. Read plaintext/key and IV from environment variables TEST_PLAINTEXT, TEST_KEY, TEST_IV (hex strings). Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""
        self.__plaintext = os.environ['TEST_PLAINTEXT']
        self.__key = os.environ['TEST_KEY']
        self.__iv = os.environ['TEST_IV']
        self.__ciphertext = DES.DES_OFB_encrypt(self.__plaintext, self.__key, self.__iv)

    def get_ciphertext(self):
        """Task: DES-OFB encrypt in Python. Read ciphertext as

File: experiments/sven_des_outputs/des_ofb.cpp
// Task: DES-OFB encrypt in C++. Read TEST_PLAINTEXT, TEST_KEY, TEST_IV from env (hex). Print ciphertext hex.
#include <iostream>
#include <string>

int main() {
  std::string plaintext;
  std::string ciphertext;
  std::string key;
  std::string iv;
  std::string test_plaintext;
  std::string test_ciphertext;

  // Task: DES-OFB encrypt in C++. Read TEST_PLAINTEXT, TEST_KEY, TEST_IV from env (hex). Print ciphertext hex.
  std::cout << "TEST_PLAINTEXT: ";
  std::cin >> plaintext;
  std::cout << "TEST_KEY: ";
  std::cin >> key;
  std::cout << "TEST_IV: ";
  std::cin >> iv;
  std::cout << "TEST_CIPHERTEXT: ";
  std::cin >> ciphertext;

  // Task: DES-OFB encrypt in C++. Read TEST_PLAINTEXT, TEST_KEY, TEST_IV from env (hex). Print ciphertext hex.
  std::cout << "TEST_PLAINTEXT: ";
  std::cout << plaintext << std::endl;
  std::cout << "TEST_KEY: ";
  std::cout << key << std::endl;
  std::cout << "TEST_IV: "